### DISCUSSION TITLE
why is the URL of got requests showing as 'unknownhost' in AWS?

### DISCUSSION THREAD

**Discussion by mverkerk-godaddy** [sindresorhus/got#2022] [Q&A] [answered]
Hi all - i have a weird issue when using the `got` library in my AWS Lambda code (the code is used for cloudwatch-synthetics canaries, in case you're familiar). Previously, I used `node-fetch`, `request`, and others and in the AWS UI i could always see the URL of any outgoing http requests - AWS synthetics must be capturing them somehow - but now that i'm using `got`, the URLs are showing up as `https://unknownhost` ... can anybody think of a reason why this may be the case? why is this happening with 'got' but not with clients?

Thanks so much for any ideas/tips!

**Comment by sindresorhus** [comment, marked as answer]
the issue is that AWS X-Ray SDK (which CloudWatch Synthetics uses) doesn't properly extract the hostname when got passes a URL object as the first argument to `http.request()`.

got calls `http.request(url, options)` with a URL object as the first arg (line 1227 in source/core/index.ts), but AWS X-Ray's hostname extraction logic expects `options.hostname` or `options.host` to be set in the options object itself.

this is an AWS X-Ray SDK limitation, not a got bug.
